### ISSUE TITLE
Re-raising Redis errors support

### ISSUE THREAD

**Issue by clusterf** [brainix/pottery#404]
When using a Redlock with blocking `acquire` or the `syncronize` decorator, if Redis is down, the `acquire` will keep being retried until Redis is back up. This could block a program indefinitely if the Redis connection cannot be recovered.

An extra argument for `syncronize` and `Redlock` that enables re-raising Redis errors (error will be raised if the argument is set to True and if the Redis errors happen for the majority of the masters) would provide better control so that the program can decide what to do in this case, instead of remaining blocked.

**Comment by brainix** [comment]
Hi, @clusterf!

Thanks for the suggestion. I've tried to incorporate it here: https://github.com/brainix/pottery/pull/406

Once I've merged that PR, you'll be able to instantiate Redlock like this:

```python
>>> redlock = Redlock(key='printer-lock', masters={redis}, raise_on_redis_errors=True)
```

Then when you try to perform any operation with that Redlock including using it as a context manager, if a majority of Redis masters throw `RedisError`s, then Pottery will re-raise a new `QuorumIsImpossible` error.

Alternatively, you can continue instantiating Redlock without the `raise_on_redis_errors` argument, but you can set that argument to `True` by calling:

```python
>>> redlock.acquire(raise_on_redis_errors=True)
>>> redlock.locked(raise_on_redis_errors=True)
>>> redlock.extend(raise_on_redis_errors=True)
>>> redlock.release(raise_on_redis_errors=True)
```

Similarly, I've added the `raise_on_redis_errors` argument to the `synchronize()` decorator.

Does this satisfy your use case? How does the PR look?

Thanks again,
Raj

**Comment by brainix** [comment]
Hi again, @clusterf.

I've just released Pottery v1.2.0, which includes this new functionality.

Thanks for the suggestion,
Raj
